201-timeseriesinsights-environment-with-eventhub/azuredeploy.json

```diff
@@ -194,16 +208,50 @@
           },
           "tags": "[variables('eventSourceTagsValue')]",
           "dependsOn": [
-            "[concat('Microsoft.TimeSeriesInsights/environments/',parameters('environmentName'))]",
+            "[concat('Microsoft.TimeSeriesInsights/environments/', parameters('environmentName'))]",
             "[concat('Microsoft.EventHub/namespaces/', parameters('eventHubNamespaceName'), '/eventHubs/', parameters('eventHubName'), '/consumerGroups/', parameters('consumerGroupname'))]"
           ]
         }
       ]
+    },
+    {
+      "condition": "[if(equals(first(parameters('accessPolicyReaderObjectIds')), 'thisArrayIsReallyEmpty'), bool('false'), bool('true'))]",
+      "type": "Microsoft.TimeSeriesInsights/environments/accesspolicies",
+      "name": "[concat(parameters('environmentName'), '/', 'readerAccessPolicy', copyIndex())]",
+      "copy": { 
+        "name": "accessPolicyReaderCopy", 
+        "count": "[if(equals(parameters('accessPolicyReaderObjectIds'), 0), 1, length(parameters('accessPolicyReaderObjectIds')))]"
```

Comment:
simplify:

length(parameters('accessPolicyReaderObjectIds'))

Reply:
Addressed. I think this must have been left over from earlier attempts to work around the copy.length problem.